test/resvg_test.exs

```diff
@@ -112,13 +139,86 @@ defmodule Resvg.Test do
       """
 
       output = image_path("cloud-from-buf.png")
+      reference = image_path("cloud.png")
+
       assert {:ok, buffer} = Resvg.svg_string_to_png_buffer(svg_string, resources_dir: @tmp)
       :ok = File.write!(output, buffer)
       assert File.exists?(output)
-      assert md5(output) == "d8407a3546efe75af450f2625f06d574"
+
+      approve snapshot: File.read!(output),
+              reference: File.read!(reference),
+              reviewed: true
     end
   end
 
+  describe "revg deals correctly with <tspan> elements inside a <text> element" do
+    # NOTE (tmbb)
+    # This is a minimal reproducible test case for a bug I found in resvg before v0.40.
+    # I don't know why this specific example triggered the bug, but it's a useful test
+    # case to keep in case there is some regression in Resvg
+
+    # Beacuse of repetitiveness in the rust rendering functions,
+    # we test the behaviour in all of them.
+    test "- function svg_string_to_png/3" do
+      input = image_path("text-font-change.svg")
+      output = image_path("snapshots/text-font-change_svg_string_to_png.png")
+      reference = image_path("text-font-change-reference.png")
+
+      svg_string = File.read!(input)
+
+      :ok = Resvg.svg_string_to_png(svg_string, output,
+        dpi: 256,
+        skip_system_fonts: true,
+        resources_dir: @tmp,
+        font_dirs: [font_dir()]
+      )
+
+      approve snapshot: File.read!(output),
+              reference: File.read!(reference),
+              reviewed: true
+    end
+
+    test "- function svg_string_to_png_buffer/3" do
+      input = image_path("text-font-change.svg")
+      output = image_path("snapshots/text-font-change_svg_string_to_png_buffer.png")
+      reference = image_path("text-font-change-reference.png")
+
+      svg_string = File.read!(input)
+
+      {:ok, image_data} = Resvg.svg_string_to_png_buffer(svg_string,
+        dpi: 256,
+        skip_system_fonts: true,
+        resources_dir: @tmp,
+        font_dirs: [font_dir()]
+      )
+
+      # Write it out becuase it's easier
+      File.write!(output, image_data)
+
+      approve snapshot: File.read!(output),
```

Comment:
It's so that you have a file you can review for differences. The `approve` macro reads the AST and expects the `File.read!` call. This is so that I can distinguish between data that comes from a file and data that might be inserted literally in the code, like other packages already do.